src/blueprint.zig

```diff
@@ -336,26 +339,30 @@ pub const Pattern = struct {
 	};
 
 	pub fn initFromString(allocator: NeverFailingAllocator, source: []const u8) !@This() {
-		var specifiers = std.mem.splitScalar(u8, source, ',');
+		var specifiers = std.mem.splitScalar(u8, source, expressionSeparator);
 		var totalWeight: f32 = 0;
 
 		var weightedEntries: ListUnmanaged(struct {block: Block, weight: f32}) = .{};
 		defer weightedEntries.deinit(main.stackAllocator);
 
 		while(specifiers.next()) |specifier| {
-			var iterator = std.mem.splitScalar(u8, specifier, '%');
+			var blockId = specifier;
+			var weight: f32 = 1.0;
 
-			var weight: f32 = undefined;
-			var block = main.blocks.parseBlock(iterator.rest());
+			if(std.mem.containsAtLeastScalar(u8, specifier, 1, weightSeparator)) {
+				var iterator = std.mem.splitScalar(u8, specifier, '%');
+				const weightString = iterator.next() orelse return error.MissingWeight;
+				blockId = iterator.next() orelse return error.MissingBlockId;
 
-			const first = iterator.first();
+				if(iterator.next() != null) return error.TooManyElements;
```

Comment:
Not needed, will be catched by invalid id path.

Reply:
Giving rather poor error message.